packages/shared/src/server/automations/webhooks.ts

```diff
@@ -0,0 +1,265 @@
+      );
+      throw new InternalServerError(
+        "Webhook config has no secret key, failing webhook execution",
+      );
+    }
+
+    if (webhookConfig.secretKey) {
+      try {
+        const decryptedSecret = decrypt(webhookConfig.secretKey);
+
+        const signature = createSignatureHeader(
+          webhookPayload,
+          decryptedSecret,
+        );
+        requestHeaders["Langfuse-Signature"] = signature;
+      } catch (error) {
+        logger.error(
+          "Failed to decrypt webhook secret or generate signature",
+          error,
+        );
+        throw new InternalServerError("Failed to generate webhook signature");
+      }
+    }
+
+    await backOff(
+      async () => {
+        logger.debug(
+          `Sending webhook to ${webhookConfig.url} with payload ${JSON.stringify(
+            webhookPayload,
+          )} and headers ${JSON.stringify(requestHeaders)}`,
+        );
+        const res = await fetch(webhookConfig.url, {
+          method: "POST",
+          body: webhookPayload,
+          headers: requestHeaders,
+        });
+
+        httpStatus = res.status;
+        responseBody = await res.text();
+
+        if (res.status !== 200) {
+          logger.error(
+            `Webhook for project ${projectId} failed with status ${res.status} and response ${responseBody}`,
+          );
+          throw new Error(
+            `Webhook for project ${projectId} failed with status ${res.status}`,
+          );
+        }
+      },
+      {
+        numOfAttempts: 4,
+      },
+    );
+
+    // Update action execution status on success
+    await prisma.actionExecution.update({
+      where: {
+        projectId,
+        triggerId: automation.trigger.id,
+        actionId: automation.action.id,
+        id: executionId,
+      },
+      data: {
+        status: ActionExecutionStatus.COMPLETED,
+        startedAt: executionStart,
+        finishedAt: new Date(),
+        output: {
+          httpStatus,
+          responseBody: responseBody?.substring(0, 1000), // Limit response body size
+        },
+      },
+    });
+
+    logger.debug(
+      `Webhook executed successfully for action ${automation.action.id}`,
+    );
+  } catch (error) {
+    logger.error("Error executing webhook", error);
+
+    const automation = await getAutomationById({
+      projectId,
+      automationId,
+    });
+
+    if (!automation) {
+      throw new LangfuseNotFoundError(`Automation ${automationId} not found`);
+    }
+
+    const shouldRetryJob =
+      error instanceof LangfuseNotFoundError ||
+      error instanceof InternalServerError;
+
+    if (shouldRetryJob) {
+      logger.warn(
+        `Retrying bullmq for webhook job for action ${automation.action.id}`,
+      );
+      throw error;
+    }
+
+    // Update action execution status and check if we should disable trigger
+    await prisma.$transaction(async (tx) => {
```

Comment:
IMO no need to wrap this in a transaction. Setting to error and then disabling can be two separate operations.